blog/_posts/2020-01-15-affiliations-feature.md

```diff
@@ -0,0 +1,80 @@
+		 	 	 		
+**Automated Pipelines: assigning identifiers to specific author-affiliation-bibcode entities**
+
+With the initial set of affiliation strings identified we, along with ADS backend programmer Stephen McDonald, designed an automated pipeline to add canonical affiliation identifiers to our database of 15 million references.  It's a simple process.  First we normalize all of the strings with all uppercase letters, remove spaces and a subset of punctuation marks not needed for disambiguation, which results in a reduced number of matching strings; often, strings will be identical without punctuation, such as strings with "... U.S.A." and "... USA".  We then assign the strings to a dictionary, where the strings themselves are the keys, and the identifiers are the values.  If a normalized incoming string matches a key in the dictionary, we assign it the appropriate aff_id, and the record with augmented affiliation data is sent back to our database of metadata for use in the next reindexing process.
+
+We've entirely automated the process, so once the dictionary of affiliation strings and matched identifiers is created, it's a hands-off component of our metadata processing pipeline.  We update dictionaries about once a month, and occasionally more often if we have a large batch of incoming unmatched affiliation strings.  This process is similar to the first curation step, with the only difference being that we prioritize strings according to how many times they appear.  For example, if we receive a number of papers from collaborations with hundreds or thousands of authors *and* their affiliations don't exactly match our dictionary, we need to add them.  Early on in this process, we came across strings that had many thousands (and sometimes tens of thousands) of occurrences, but the latest batch of unclassified strings appear fewer than 550 times in our entire database. This maximum frequency drops each time we create a new dictionary.
+
+However, we're getting close to the limit of what we can do with human classification without lots of effort.  Our most recent pass through our metadata has just over 4.6 million unclassified strings.  There are about 6000 unclassified affiliations that appear 100 times or more in the metadata and assigning them aff_ids would provide a lot of new information.  However, over 65 percent of unclassified affiliations occur fewer than 10 times, and **13 percent of unmatched affiliation strings only appear once in all of our affiliation metadata**.
```

Comment:
"...4.6 million unclassified strings" - but the top of the right panel of the figure below says 16.6 million unmatched affiliation strings?

Reply:
I clarified this -- it's 4.8 million *unique* strings  which yields a total of 16.6 million total unmapped affiliations.